### DISCUSSION TITLE
bitmap "favicon.ico" not defined exception

### DISCUSSION THREAD

**Discussion by Mathiaszero** [pyinstaller/pyinstaller#8913] [Help]
![image](https://github.com/user-attachments/assets/9c9b053f-c0c1-4767-b194-cd5e7f7c8fb5)

So in tkinter I set icon and when I run the app from the .py file it works fine; pyinstaller also builds the .exe fine. However when I run it I get the issue above. I have also toggled the Sun Valley theme extension. How can I fix this?

**Comment by rokm** [comment]
You need to load your icon using absolute path instead of just relative basename; for example, by anchoring it to parent of `__file__`, as described in the [docs](https://pyinstaller.org/en/stable/runtime-information.html#using-file).

For example, assuming line 28 in `Window.py` is currently
```python
root.iconbitmap(default='favicon.ico')
```
it should be changed to
```python
root.iconbitmap(default=os.path.join(os.path.dirname(__file__), 'favicon.ico'))
```
(that's assuming that the icon is located in same directory as `Window.py`).

And you should also ensure that the icon is collected in the frozen application, for example, by using [`--add-data favicon.ico:.`](https://pyinstaller.org/en/stable/usage.html#cmdoption-add-data).
